experimental/AlgebraicStatistics/src/serialization.jl

```diff
@@ -0,0 +1,41 @@
+@register_serialization_type PhylogeneticModel 
+@register_serialization_type GroupBasedPhylogeneticModel 
+
+function save_object(s::SerializerState, pm::PhylogeneticModel)
+  save_data_dict(s) do 
+    save_object(s, graph(pm), :tree) # already implemented 
+    save_object(s, number_states(pm), :states) # already implemented 
+    save_object(s, probability_ring(pm), :probability_ring) # already implemented
+    save_object(s, root_distribution(pm), :root_distribution) # needs to be implemented, or we change the entry type to QQFieldElem
+    save_object(s, transition_matrices(pm), :transition_matrices) # needs to be implemented
+  end
+end
+
+function save_object(s::SerializerState, pm::GroupBasedPhylogeneticModel)
+  save_data_dict(s) do
+    save_object(s, phylogenetic_model(pm), :phylomodel) # details see above
+    save_object(s, fourier_ring(pm), :fourier_ring) # already implemented
+    save_object(s, fourier_parameters(pm), :fourier_params) # needs to be implemented
+    save_object(s,group_of_model(pm), :group) # already implemented
+  end
+end
+
+
+function load_object(s::DeserializerState, ::Type{<: PhylogeneticModel})
+  graph = load_object(s, Graph{Directed}, :tree)
+  n_states = load_object(s, Int64, :states)
+  prob_ring = load_object(s, QQMPolyRing, :probability_ring)
```

Comment:
this would then become
```suggestion
  prob_ring = load_typed_object(s, :probability_ring)
```